comp/core/tagger/collectors/workloadmeta_main.go

```diff
@@ -103,30 +103,39 @@ func (c *WorkloadMetaCollector) collectStaticGlobalTags(ctx context.Context) {
 		// to read it, for the instances where we are running in an environment where hostname cannot be detected.
 		if cluster := clustername.GetClusterNameTagValue(ctx, ""); cluster != "" {
 			if c.staticTags == nil {
-				c.staticTags = make(map[string]string, 1)
+				c.staticTags = make(map[string][]string, 1)
 			}
-			c.staticTags[clusterTagNamePrefix] = cluster
+			if _, exists := c.staticTags[clusterTagNamePrefix]; !exists {
+				c.staticTags[clusterTagNamePrefix] = []string{}
+			}
+			c.staticTags[clusterTagNamePrefix] = append(c.staticTags[clusterTagNamePrefix], cluster)
 		}
 	}
-	if len(c.staticTags) > 0 {
-		tags := taglist.NewTagList()
+	// These are the global tags that should only be applied to the internal global entity
+	// Whereas the static tags are applied to containers and pods directly as well.
+	globalEnvTags := util.GetGlobalEnvTags()
```

Comment:
Could we pass the config component to `util.GetGlobalEnvTags()`? That was in the body of that function do not access the global config object and instead use the config component 😄

Reply:
Yes I can definitely pass down the tagger's reference to the `config.Component` to this function. However, another thing to note is that the `GetStaticTags` and `GetStaticTagsSlice` that is also originally used in this function and also the `dogstatsd/server/server.go` ([ref](https://github.com/DataDog/datadog-agent/blob/7e3ee88ba04ad1760ce5445ed5bbdfb257f73aa4/comp/dogstatsd/server/server.go#L239)) are using the global config object too. I can cut a ticket to work on later to also have these helper functions use the config Component.